### PR TITLE
Beta: Add Jetpack Constants

### PR DESCRIPTION
**Pull request by kraftbj**
Replaces https://github.com/Automattic/jetpack-beta/pull/72

Add the constants settings page here instead of the JN companion plugin.

Right now, when asking folks to test things like beta blocks, there is no built-in way for beta testers to do this without manually adjusting config files.

TODO: The review comment left in https://github.com/Automattic/jetpack-beta/pull/72#pullrequestreview-566956273

### CODE REVIEW

**Review by jeherve — CHANGES_REQUESTED** [review]
I think you'll need to move the `lib/RationalOptionPages.php` inside the plugin's folder.

**Comment by kraftbj** [comment]
Thanks, I missed that when trying to merge up the branch from the old repo.

**Comment by kraftbj** [comment]
I'm going to parse out separately improving this. I'm not a fan of the RationalOptionsPages tool used for the options page of Constants, so I'll want to rework that.

File: projects/plugins/beta/lib/RationalOptionPages.php [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+					$sort_order = array_keys( $this->defaults['add_settings_section'] );
+					$params = $this->sort_array( $section_params, $sort_order );
+					$params = array_slice( $params, 0, count( $this->defaults['add_settings_section'] ) );
+
+					// Finalize callback
+					if ( empty( $params['custom'] ) && !is_array( $params['callback'] ) ) {
+						$params['callback'] = array( $this, $params['callback'] );
+					}
+
+					call_user_func_array( 'add_settings_section', $params );
```
Comment (jeherve) [inline projects/plugins/beta/lib/RationalOptionPages.php:256]: Let's bring in the changes from https://github.com/Automattic/companion/pull/44

**Review by jeherve — CHANGES_REQUESTED** [review]

**Pull request closed without merging**